### PR TITLE
Ah add sherlock

### PR DESCRIPTION
**Pull request by ahaessly**
update github actions to integrate with sherlock

### CODE REVIEW

File: .github/workflows/build_publish.yml [hunk 1]
```diff
@@ -77,14 +83,27 @@ jobs:
...
+    # Report new cancerdata version to Broad DevOps
+    uses: broadinstitute/sherlock/.github/workflows/client-report-app-version.yaml@main
+    needs: [ tag-build-publish ]
+    with:
+      new-version: ${{ needs.tag-build-publish.outputs.tag }}
+      chart-name: 'cancerdata'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+#    inputs: '{ "bee-name": "${{ github.event.repository.name }}-${{ github.run_id }}-${{ matrix.terra-env }}" }'
```
Comment (bahill) [inline .github/workflows/build_publish.yml:96]: Is there a reason to keep this comment in? NBD, just wondering from a hygiene perspective.

File: .github/workflows/master_push.yml
Comment (bahill) [inline .github/workflows/master_push.yml:1]: Again, just curious. Why remove this file?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -27,11 +27,6 @@ jobs:
         restore-keys: ${{ runner.os }}-gradle
     - name: Run tests
       run: ./gradlew test --scan
-    - name: Run SonarQube scan
```
Comment (bahill) [inline .github/workflows/tests.yml:30]: Last bit of curiosity - why remove SonarQube?

File: .github/workflows/build_publish.yml (hunk 1 above)
Comment (ahaessly) [reply .github/workflows/build_publish.yml:96]: I'll remove

File: .github/workflows/tests.yml (hunk 2 above)
Comment (ahaessly) [reply .github/workflows/tests.yml:30]: I no longer supports java 11. Info sec has added a ticket to our board to replace sonarqube with another tool

File: .github/workflows/master_push.yml
Comment (ahaessly) [reply .github/workflows/master_push.yml:1]: there was too much duplication in master_push and build_publish and I just happened to work on the other file first. Also we renamed out default branch to main.

**Review by bahill — APPROVED** [review]
lgtm!

**Review by fthomas641 — APPROVED** [review]

**Review by abradyGDIT — APPROVED** [review]

**Pull request merged**